project-docs/stumptown-writers-guide.md

```diff
@@ -0,0 +1,697 @@
+
+The H1 heading must be in `code format` and its content must be the name of the attribute:
+
+```md
+# `crossorigin`
+```
+
+Prose immediately after this describes the attribute.
+
+Some attributes, such as the `crossorigin` attribute above, can take special enumerated values. For such attributes, the description should be followed by an H2 with the text "Values":
+
+```
+## Values
+```
+
+If there is a "Values" H2, then it must be followed by one or more H3 headings, with text content in `code format`, whose text content is a possible value for the attribute.
+
+The text following this H3 describes that value.
+
+After the "Values" section there is a mandatory H2 "Type":
+
+```
+## Type
+```
+
+After this there is an string describing the type of this attribute. The following are valid values for this string:
+
+"String", "Number", "Boolean", "URL", "RegExp".
+
+### examples
+
+An object used to specify static or live samples to include in the page.
+
+**Note: this does not specify interactive examples, because we'd have to rewrite the live sample system if we wanted to do that.**
+
+It is given as an array of paths to directories. Each directory contains a single example:
+
+```yaml
+examples:
+    - examples/simple-example
+    - examples/audio-element-with-video-element
+    - examples/multiple-source-elements
+```
+
+**Note: it looks like this is using a different path convention from attributes. We should be consistent here.**
+
+Examples will be included in the rendered output in the order given here: so for example you may want to present the basic example first in the array, then more complex ones.
+
+Each example is specified as a directory containing four files, one mandatory and three optional:
+
+-   "description.md" - the only mandatory one
+-   "example.html"
+-   "example.css"
+-   "example.js"
+
+**Note: we don't yet support "hidden" sources One suggestion was just to call them "hidden.css" etc.**
+
+#### description.md
+
+A Markdown file containing some YAML front matter.
+
+The front matter contains a mandatory `title` and an optional `height`:
+
+```
+---
+title: Custom error messages
+height: 192
+---
+```
+
+If `height` is included, then stumptown treats this example as a live sample, and `height` is the height of the "Result" iframe in pixels (the iframe always gets full page width).
+
+If `height` is omitted, then stumptown treats this example as just a static example.
+
+**Note: many examples currently omit "title".**
+
+**Note: we have also considered having a less implicit way to signal whether or not an example is live (https://github.com/mdn/stumptown-content/issues/45 ).**
+
+The rest of the "description.md" file is a prose description of the example.
+
+#### example.\*
+
+The example.html, example.css, and example.js files contain the example sources themselves.
+
+### specifications
+
+This property lists specifications for the item.
+
+This is given as one of:
```

Comment:
Two points:

* What about IETF specs in RFCs? We don't currently list these in our spec database, but do sometimes link to them in spec tables. We should decide how to handle these and then write that down in here.

* We should have some written rule about the boundary between what's got an entry in the spec table and what doesn't. We do sometimes document things which are "specified" in non-official documents such as community drafts. For instance, there has been discussion about providing some documentation for things like EPUB (https://www.w3.org/publishing/epub32/epub-spec.html), whose spec is a community group publication rather than an "official" W3C spec.

Reply:
@a2sheppy , please file these as separate issues.